examples/02_explaining_similarity_encoding.py

```diff
@@ -0,0 +1,148 @@
+"""
+.. _example_explaining_similarity_encoding:
+
+===================================
+Investigating the SimilarityEncoder
+===================================
+
+In this example, we will take a deeper look at how the |SimilarityEncoder|
+works, and how it compares to the |OneHotEncoder| and other encoders in the
+library.
+
+
+.. |OneHotEncoder| replace::
+    :class:`~sklearn.preprocessing.OneHotEncoder`
+
+.. |SimilarityEncoder| replace::
+    :class:`~skrub.SimilarityEncoder`
+"""
+
+###############################################################################
+# What do we mean by dirty categories?
+# ------------------------------------
+#
+# Let's get the table we will use in this example,
+# the `employee salaries dataset <https://www.openml.org/d/42125>`_:
+
+from skrub.datasets import fetch_employee_salaries
+
+dataset = fetch_employee_salaries()
+print(dataset.description)
+
+X = dataset.X
+y = dataset.y
+
+X.head()
+
+
+###############################################################################
+# These different entries are often variations of the same entity.
+# For example, there are 3 kinds of "Accountant/Auditor".
+#
+# Such variations will break traditional categorical encoding methods:
+#
+# * Using a simple |OneHotEncoder|
+#   will create orthogonal features, whereas it is clear that
+#   those 3 terms have a lot in common.
+#
+# * If we wanted to use word embedding methods such as
+#   `Word2vec <https://www.tensorflow.org/tutorials/text/word2vec>`_,
+#   we would have to go through a cleaning phase: those algorithms
+#   are not trained to work on data such as "Accountant/Auditor I".
+#   However, this can be error-prone and time-consuming.
+#
+# The problem becomes easier if we can capture relationships between
+# entries.
+#
+# To simplify understanding, we will focus on the column describing the
+# employee's position title:
+
+values = X[["employee_position_title", "gender"]]
+values.insert(0, "current_annual_salary", dataset.y)
+
+sorted_values = values["employee_position_title"].sort_values().unique()
+
+###############################################################################
+# In order to robustly embed dirty semantic data, the |SimilarityEncoder|
+# creates a similarity matrix based on an n-gram representation of the data.
+
+from skrub import SimilarityEncoder
+
+similarity_encoder = SimilarityEncoder()
+transformed_values = similarity_encoder.fit_transform(sorted_values.reshape(-1, 1))
```

Comment:
Print a small subset of the similarity matrix to show what it looks like?
You can quickly state below that the values are in between 0 and 1, based on the level of n-gram similarity.